src/utils/compare.ts

```diff
@@ -6,7 +6,15 @@ export const equalColorObjects = (first: HSL | HSV | RGB, second: HSL | HSV | RG
   if (first === second) return true;
 
   for (const prop in first) {
-    if (first[prop] !== second[prop]) return false;
+    // The following allows for a type-safe calling of this function (first & second have to be HSL, HSV, or RGB)
+    // with type-unsafe iterating over object keys. TS does not allow this without an index (`[key: string]: number`)
+    // on an object to define how iteration is normally done. To ensure extra keys are not allowed on our types,
+    // we must cast our object to unknown, and then as a type TS can iterate over.
+    if (
+      ((first as unknown) as Record<string, number>)[prop] !==
+      ((second as unknown) as Record<string, number>)[prop]
+    )
```

Comment:
@omgovich Is this along the lines of what you were thinking of?

Reply:
Yeah. Now everyone will be able to understand the magic that we're doing here =)

Reply:
Just added a bit more to why we need to first cast to `unknown` as that was lacking.

Reply:
❤️